From e83ec930ca49b6a2ab2ab2093b04b7362449c4ce Mon Sep 17 00:00:00 2001
From: Hannes Reinecke <hare@suse.de>
Date: Wed, 6 Dec 2017 11:27:54 +0100
Subject: [PATCH] Disable
 patches.kernel.org/4.4.93-022-fix-unbalanced-page-refcounting-in-bio_map_use.patch
 (bsc#1070767)

References: bsc#1070767
Patch-mainline: no, hare specific

Refresh patches.drivers/0004-bio-use-offset_in_page-macro.patch.

suse-commit: ad107d81cba4f3b0c870f47ecb665ef1224b5985
Acked-by: Michal Suchanek <msuchanek@suse.de>
---
 block/bio.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/block/bio.c b/block/bio.c
index 7702662a1270..08d86f2d527d 100644
--- a/block/bio.c
+++ b/block/bio.c
@@ -1378,7 +1378,6 @@ struct bio *bio_map_user_iov(struct request_queue *q,
 		offset = offset_in_page(uaddr);
 		for (j = cur_page; j < page_limit; j++) {
 			unsigned int bytes = PAGE_SIZE - offset;
-			unsigned short prev_bi_vcnt = bio->bi_vcnt;
 
 			if (len <= 0)
 				break;
@@ -1393,13 +1392,6 @@ struct bio *bio_map_user_iov(struct request_queue *q,
 					    bytes)
 				break;
 
-			/*
-			 * check if vector was merged with previous
-			 * drop page reference if needed
-			 */
-			if (bio->bi_vcnt == prev_bi_vcnt)
-				put_page(pages[j]);
-
 			len -= bytes;
 			offset = 0;
 		}
-- 
2.20.1

